test/rql_test/connections/twisted_connection.py

```diff
@@ -0,0 +1,950 @@
+        yield done
+
+class TestChangefeeds(TestWithConnection):
+    @inlineCallbacks
+    def setUp(self):
+        yield TestWithConnection.setUp(self)
+        self.conn = yield r.connect(host=sharedServerHost,
+                                    port=sharedServerDriverPort)
+        if 'a' in (yield r.db('test').table_list().run(self.conn)):
+            yield r.db('test').table_drop('a').run(self.conn)
+        yield r.db('test').table_create('a').run(self.conn)
+        if 'b' in (yield r.db('test').table_list().run(self.conn)):
+            yield r.db('test').table_drop('b').run(self.conn)
+        yield r.db('test').table_create('b').run(self.conn)
+
+    @inlineCallbacks
+    def tearDown(self):
+        yield r.db('test').table_drop('b').run(self.conn)
+        yield r.db('test').table_drop('a').run(self.conn)
+        yield TestWithConnection.tearDown(self)
+        yield self.conn.close()
+        self.conn = None
+
+    @inlineCallbacks
+    def table_a_even_writer(self):
+        for i in range(10):
+            yield r.db('test').table("a").insert({"id": i * 2}).run(self.conn)
+
+    @inlineCallbacks
+    def table_a_odd_writer(self):
+        for i in range(10):
+            yield r.db('test').table("a").insert({"id": i * 2 + 1}).run(self.conn)
+
+    @inlineCallbacks
+    def table_b_writer(self):
+        for i in range(10):
+            yield r.db('test').table("b").insert({"id": i}).run(self.conn)
+
+    @inlineCallbacks
+    def cfeed_noticer(self, table, ready, done, needed_values):
+        feed = yield r.db('test').table(table).changes(squash=False).run(self.conn)
+        try:
+            ready.callback(None)
+            for item in feed:
+                if len(needed_values) != 0:
+                    break
+
+                item = yield feed.next()
+                self.assertIsNone(item['old_val'])
+                self.assertIn(item['new_val']['id'], needed_values)
+                needed_values.remove(item['new_val']['id'])
+            done.callback(None)
+        except Exception as ex:
+            done.errback(ex)
+
+    @inlineCallbacks
+    def test_multiple_changefeeds(self):
+        feeds_ready = {}
+        feeds_done = {}
+        needed_values = { 'a': set(range(20)), 'b': set(range(10)) }
+        for n in ('a', 'b'):
+            feeds_ready[n] = defer.Deferred()
+            feeds_done[n] = defer.Deferred()
+            reactor.callInThread(self.cfeed_noticer, n, feeds_ready[n],
+                                 feeds_done[n], needed_values[n])
+
+        yield defer.DeferredList(feeds_ready.values())
+        yield defer.DeferredList([self.table_a_even_writer(),
+               self.table_a_odd_writer(),
+               self.table_b_writer()])
+        yield defer.DeferredList(feeds_done.values())
+        self.assertTrue(all([len(x) == 0 for x in needed_values.values()]))
+
+class AsyncSuite(unittest.TestSuite):
+
+    def _destroyReactor(self):
+        from twisted.internet import reactor
```

Comment:
`twisted.internet.reactor` is already imported in this file - not sure how this might affect that, but it looks bad